Add more virtual return fixes.

tbaker57 opened the pull request:
More fixes for the December Beginner sprint. With @abednarik 's help we might finish the sprint before Christmas :smile:

kiorky committed:
Fix #29742: vgname is lost in the middle

moali87 committed:
Adding message for null public IP

terminalmage committed:
git.latest: init submodules if not yet initialized

Fixes #29009.

Commit:
Refactored ESXCLI-based functions to accept a list of esxi_hosts

Previously, only one esxi_host could be passed in when connecting
to a vCenter server. This brings feature-partiy for ESXCLI-based
functions with pyVmomo-based functions, which already accept and
handle a list of hosts.

State functions were also updated.

Also fixes a bug with the ESXCLI which check in salt.utils.vmware
and provides default protocol and port options if none are provided.

ruxandraburtica committed:
Returning security group when no VPC id is given, even if the group is not in EC2-classic.

bdrung committed:
Fix typo comparsion -> comparison

bdrung committed:
Fix typo specfic -> specific

joehealy committed:
Minor update to release notes for missing fix

Signed-off-by: Benjamin Drung <benjamin.drung@profitbricks.com>

anlutro committed:
fix a call to deprecated method in python-influxdb

https://github.com/influxdb/influxdb-python/blob/1da39bf8603e7bab2d1bc6424996a0a90560449f/influxdb/influxdb08/client.py#L543-L553

anlutro committed:
update test

Commit:
Merge pull request #29565 from bdrung/2015.8

Fix typos and missing release note

Commit:
Merge pull request #29563 from alprs/fix-influx_deprecated_method

Fix a call to deprecated method in python-influxdb

Commit:
Merge pull request #29547 from rallytime/esxi-proxy

Refactored ESXCLI-based functions to accept a list of esxi_hosts

Commit:
Merge pull request #29548 from rallytime/bp-29449

Back-port #29449 to 2015.8

Commit:
Merge pull request #29545 from terminalmage/issue29009

git.latest: init submodules if not yet initialized

gtmanfred committed:
use old discover_extensions if available

This was moved in 2.24.0, but this should work for everything back to 2.6.1

Closes #29557

Commit:
Merge pull request #29572 from gtmanfred/2015.8

[nova] use old discover_extensions if available

ruxandraburtica committed:
Replaced tabs with spaces.

job committed:
Fix the 'cidr' arg in salt.modules.network.ip_addrs6()

The two arguments to ip_in_subnet() should be swapped around, but
since ip_in_subnet() is being deprecated in favor of in_subnet(),
removing three characters seemed like the most appropiate bugfix.

This fixes #29585

Commit:
Added ESXi Proxy Minion Tutorial

Also added some more documentation to other ESXi-related files
and fixed some related doc errors.

Commit:
Merge pull request #29588 from rallytime/esxi-proxy-tutorial

Added ESXi Proxy Minion Tutorial

terminalmage committed:
dockerng: Prevent exception when API response contains empty dictionary

Fixes #29584.

basepi committed:
Fix error in failure checking for salt.state within orchestration

basepi committed:
Remove unnecessary and

Commit:
Merge pull request #29596 from rallytime/bp-29587

Back-port #29587 to 2015.8

Commit:
Merge pull request #29597 from terminalmage/issue29584

dockerng: Prevent exception when API response contains empty dictionary

abednarik committed:
Fixed duplicate mtu entry in RedHat 7  network configuration.

Removed duplicate mtu entry in jinja template for RedHat 7 Systems.

basepi committed:
Merge pull request #29603 from basepi/orchestrate.failures.29546

Fix orchestration failure-checking

basepi committed:
Allow setting the cmd_umask from within the roster

basepi committed:
Add cmd_umask to roster docs

Commit:
Merge pull request #29609 from basepi/salt-ssh.umask.29574

[2015.8][salt-ssh] Add ability to set salt-ssh command umask in roster

Commit:
Various ESXi Proxy Minion Bug Fixes

Including, but not limited to:

- Quote ESXCLI password to handle password spaces
- Only return vsan disk names instead of disk objects for vsan funcs
- Add warning to quote "on" and "off" service policy states
- Catch vmodl.fault.SystemError Errors when resetting passwords

Commit:
Don't create io_loop before fork

Commit:
Merge pull request #29628 from DSRCompany/bug/26364_freebsd_tcp

Don't create io_loop before fork

Commit:
Merge pull request #29613 from rallytime/esxi-fixes

Various ESXi Proxy Minion Bug Fixes

Commit:
Merge pull request #29606 from abednarik/remove_duplicate_mtu_entry_rh7_net_template

Fixed duplicate mtu entry in RedHat 7  network configuration.

jfindlay committed:
modules.tls.__virtual__: remove redundant parens

jfindlay committed:
modules.tls.__virtual__: refactor cert path comment

jfindlay committed:
modules.tls.__virtual__: don't spam everyone's error log

Commit:
Merge pull request #29632 from jfindlay/tls_virt

reduce severity of tls module __virtual__ logging

Commit:
Fixed a couple more ESXi proxy minion bugs

- Don't log the salt.utils.vmware.esxcli command (shouldn't log password)
- Make sure we catch all ssh key changes possibilities in esxi state
- Provide a use-case example of why using a list of passwords for the host is a useful idea

thusoy committed:
Support match-sets in iptables module

Based on work by @l13t in https://github.com/saltstack/salt/pull/29426.

Added tests and fixed incorrect appending of the built rule.

Closes #29423.

serge-p committed:
Update user.py

### ISSUE: 
`salt/states/user.py` passes 3 args to virtual module user, but 
`salt/modules/mac_user.py` accepts only 2 arguments 

actual exception: 
```
 Name: lemon - Function: group.present - Result: Clean
----------
          ID: lemon
    Function: user.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/salt/state.py", line 1594, in call
                  **cdata['kwargs'])
                File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/salt/loader.py", line 1491, in wrapper
                  return f(*args, **kwargs)
                File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/salt/states/user.py", line 471, in present
                  __salt__['user.chhome'](name, val, True)
              TypeError: chhome() takes exactly 2 arguments (3 given)
     Started: 23:03:30.934322
    Duration: 643.339 ms
     Changes:   
```

###PROPOSAL: 

to check for Darwin kernel inside `states.user.py` ,  ignoring persist arg,

unicolet committed:
salt-cloud: improve linode provider to optionally create an additional datadisk

unicolet committed:
salt-cloud: fix two errors that slipped through in the linode driver

abednarik committed:
Added a note regarding logging configuration.

Simple example on how to turn off logging.

MIBc committed:
Summary: Add two neutron api about listing and showing firewall.
Description:
Add function list_firewalls and show_firewall in salt/modules/neutron.py
Add object function list_firewalls, show_firewall and _fetch_firewall in
salt/utils/openstack/neutron.py

unicolet committed:
obey pylint

jirikotlin committed:
checkrestart functionality for Debian and Red Hat Based systems

Commit:
strips stdout of every line before the last RSTR one

jirikotlin committed:
Added codeauthor and versionadded comments.

jirikotlin committed:
pylint Python 3 code fix.

Commit:
Merge pull request #29645 from rallytime/bp-29558

Back-port #29558 to 2015.8

abednarik committed:
modules.deb_postgres: __virtual__ return err msg.

Updated message in deb_postgres module when return False if pg_createcluste command is not available.

Commit:
Merge pull request #29644 from rallytime/esxi-fixes

Fixed a couple more ESXi proxy minion bugs

basepi committed:
Merge remote-tracking branch 'upstream/2015.8' into merge-forward-develop

Conflicts:
    salt/client/ssh/ssh_py_shim.py
    salt/modules/tls.py
    salt/states/git.py
    salt/transport/tcp.py

Commit:
Deprecate htpasswd useradd_all in favor of useradd

Fixes #29623

One function is just a wrapper for the other. There is no need to
have both, and having both is confusing, especially since the
documentation for both of these functions is the same.

lomeroe committed:
Allow windows minion to run beacon
add documentation to show usage for windows drive letters

serge-p committed:
Update user.py

abednarik committed:
modules.lxc: __virtual__ return err msg.

Updated message in lxc module when return False if lxc-start command is not available.

abednarik committed:
modules.portage_config: __virtual__ return err msg.

Updated message in portage_config module when return False if portage is not available or os
is not Gentto.

unicolet committed:
documentation: add versionadded to new functions

Commit:
ssh client: remove lines before any RSTR in stdout/stderr

jirikotlin committed:
salt.utils.fopen instead of python open

Commit:
ssh client: fix shim_cmd access denied

abednarik committed:
modules.marathon: __virtual__ return err msg.

Updated message in marathon  module when return False if is not used in proxy minions.

abednarik committed:
modules.random: __virtual__ return err msg.

Updated message in mod_random module when return False if Python version is not 2.7 or higher.

bradthurber committed:
simplify profile/no profile logic to avoid repetition, add additional "LONG" datatype token, return stdout in cli_result as some state error messages depend on it being there

terminalmage committed:
Add optional "info" param to CommandExecutionError

terminalmage committed:
aptpkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
brew.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

lomeroe committed:
removing unused import to make lint happy

terminalmage committed:
ebuild.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
freebsdpkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
macports.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
openbsdpkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
opkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
pacman.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
pkgin.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

Commit:
Pylint for #29653

Refs #29653

Commit:
Merge pull request #29709 from rallytime/merge-29653

Merge #29653 with pylint fix

Commit:
Merge pull request #29705 from abednarik/virtual_ret_mod_randon_module

modules.random: __virtual__ return err msg.

Commit:
Merge pull request #29704 from abednarik/virtual_ret_marathon_module

modules.marathon: __virtual__ return err msg.

Commit:
Merge pull request #29695 from abednarik/virtual_ret_lxc_module

modules.lxc: __virtual__ return err msg.

techhat committed:
Add qemu_img.convert

bradthurber committed:
fix whitespace lint

Commit:
Check for windows dropfile existence

This will check the existence of the .dfn dropfile before attempting to
change its permissions to avoid the exception, "The system cannot find
the file specified" in the case when the file does not exist.

Signed-off-by: Perry Zipoy <perry.zipoy@ni.com>

techhat committed:
Add genesis.mksls

Commit:
Merge pull request #29710 from techhat/imgconvert

Add qemu_img.convert

techhat committed:
Linting

Commit:
Merge pull request #29692 from lomeroe/diskusage_beacon_windows

Allow windows minion to run beacon

Commit:
Merge pull request #29696 from abednarik/virtual_ret_portage_config_module

modules.portage_config: __virtual__ return err msg.

Commit:
Merge pull request #29689 from rallytime/fix-29623

Deprecate htpasswd useradd_all in favor of useradd

Commit:
Merge pull request #29688 from basepi/merge-forward-develop

Merge forward from 2015.8 to develop

basepi committed:
Merge pull request #29703 from dr4Ke/fix_ssh_shim_RSTR_twice_in_stdout

Fix ssh shim rstr twice in stdout

Commit:
Merge pull request #29680 from abednarik/virtual_ret_deb_postgresql_module

modules.deb_postgres: __virtual__ return err msg.

basepi committed:
Merge pull request #29702 from dr4Ke/fix_ssh_shim_local_temp_access_denied

ssh client: fix shim_cmd access denied

Commit:
Merge pull request #29655 from abednarik/disable_log_doc_update

Added a note regarding logging configuration.

Commit:
Merge pull request #29647 from serge-p/patch-4

Update user.py

techhat committed:
More lint

Commit:
Merge pull request #29706 from bradthurber/jboss-modules-updates

Jboss module minor updates

Commit:
Merge pull request #29669 from jirikotlin/develop

checkrestart functionality

jraby committed:
Document syslog facility format and default value

gqgunhed committed:
fixed: include all items from kern.disks split

salt['grains.get']('disks') always dropped the last disk (first in returned list from sysctl).
"sysctl -n" does not include sysctl-name, so the [1:] offset was wrong.

s-sebastian committed:
Remove humanname from yum repository options

terminalmage committed:
pkgng.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
solarisips.py: raise exception if changes fail, fix API non-compliance

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

It also corrects several non-compliances with Salt's pkg management API
that would cause issues with states.

terminalmage committed:
solarispkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
yumpkg.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

Also remove the now-unused ``pkg._avail`` context key, which was removed
when repoquery was removed from yumpkg.py.

terminalmage committed:
Remove unused pkg._avail context key references

terminalmage committed:
zypper.py: raise exception if changes fail

This uses the new 'info' param for the CommandExecutionError exception
to allow us to report partial changes when there are errors.

terminalmage committed:
Get changes/error message for pkg states from new CommandExecutionError attributes

Commit:
Disable some boto tests per resolution of moto issue

Pending resolution of https://github.com/spulec/moto/issues/493

Commit:
Merge pull request #29725 from cachedout/disable_moto

Disable some boto tests per resolution of moto issue

Commit:
Merge pull request #29720 from s-sebastian/develop

Remove humanname from yum repository options

Commit:
Merge pull request #29717 from jraby/patch-1

Document syslog facility format and default value

Commit:
Merge pull request #29728 from pzipoy/dev/pzipoy/windows_dropfile_exist

Check for windows dropfile existence

MIBc committed:
Update neutron.py

Granjow committed:
Doc update on masterless mode: Move master config to minion config

kiorky committed:
Retrocompat fix for #29735: pkg.latest/wait

This refs & fix salstack/salt#29735

abednarik committed:
modules.win_service: __virtual__ return err msg.

Updated messafe when return Falso on __virtual__ when system is not Windows.

kiorky committed:
Type mess in git.latest

Did not get why, but we can have a list of comments instead of stings.

This fixes:
```
     line 107, in _uptodate\n    ret['comment'] += '\\n\\nChanges made: ' + comments\nTypeError:
      cannot concatenate 'str' and 'list' objects\n"
    duration: 366.804
    name: https://github.com/makinacorpus/makina-states.git
    result: false
```

abednarik committed:
modules.win_pkg: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_path: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_ntp: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_network: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_ip: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_groupadd: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

Commit:
Merge pull request #29745 from abednarik/virtual_ret_win_pkg_module

modules.win_pkg: __virtual__ return err msg.

Commit:
Merge pull request #29750 from abednarik/virtual_ret_win_groupadd_module

modules.win_groupadd: __virtual__ return err msg.

Commit:
Merge pull request #29749 from abednarik/virtual_ret_win_ip_module

modules.win_ip: __virtual__ return err msg.

Commit:
Merge pull request #29748 from abednarik/virtual_ret_win_network_module

modules.win_network: __virtual__ return err msg.

Commit:
Merge pull request #29747 from abednarik/virtual_ret_win_ntp_module

modules.win_ntp: __virtual__ return err msg.

Commit:
Merge pull request #29746 from abednarik/virtual_ret_win_path_module

modules.win_path: __virtual__ return err msg.

Commit:
Merge pull request #29740 from kiorky/fixgit

Type mess in git.latest

Commit:
Merge pull request #29744 from kiorky/vgname

Fix #29742: vgname is lost in the middle

Commit:
Merge pull request #29739 from abednarik/virtual_ret_win_service_module

modules.win_service: __virtual__ return err msg.

Commit:
Merge pull request #29738 from kiorky/fixpkg

Retrocompat fix for #29735: pkg.latest/wait

Commit:
Merge pull request #29734 from Granjow/masterless-config

Doc update on masterless mode: Move master config to minion config

Commit:
Merge pull request #29719 from gqgunhed/gqgunhed-patch

fixed: include all items from kern.disks split

Commit:
Merge pull request #29718 from thusoy/issue-29423

Support match-sets in iptables module

Commit:
Merge pull request #29712 from techhat/mksls

Add genesis.mksls

Commit:
Merge pull request #29658 from MIBc/virt_ret

Summary: Add two neutron api about listing and showing firewall.

Commit:
Skip refresh db test if upstream repo does not repsond.

slav0nic committed:
Fix #25930

abednarik committed:
modules.win_firewall: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_file: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_dns_client: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

abednarik committed:
modules.win_disk: __virtual__ return err msg.

Updated message when return False on __virtual__ when system is not Windows.

Commit:
Add documentation about scopes (GCE)

Commit:
Merge pull request #29757 from cachedout/refresh_db

Skip refresh db test if upstream repo does not repsond.

Commit:
Splay sleeps inside jenkins

Commit:
Merge pull request #29769 from pass-by-value/gce_scopes_doc

Add documentation about scopes (GCE)

Commit:
Merge pull request #29763 from abednarik/virtual_ret_win_disk_module

modules.win_disk: __virtual__ return err msg.

Commit:
Merge pull request #29760 from abednarik/virtual_ret_win_firewall_module

modules.win_firewall: __virtual__ return err msg.

Commit:
Merge pull request #29762 from abednarik/virtual_ret_win_dns_client_module

modules.win_dns_client: __virtual__ return err msg.

Commit:
Merge pull request #29761 from abednarik/virtual_ret_win_file_module

modules.win_file: __virtual__ return err msg.

Commit:
Merge pull request #29759 from slav0nic/patch-1

Fix #25930

travispaul committed:
The crontab for SmartOS, Solaris, and AIX do not have a mechanism
to supply a user name AND a file.

Commit:
Merge pull request #29774 from cachedout/splay_jenkins

Splay sleeps inside jenkins

akissa committed:
Add Pillar SQLCipher module

Fixes #29781

abednarik committed:
modules.win_dacl: __virtual__ return err msg.

Updated message in win_dacl module when return False if is not a Windows systems.

abednarik committed:
modules.win_autoruns: __virtual__ return err msg.

Updated message in win_autoruns module when return False if is not a Windows systems.

abednarik committed:
modules.test_virtual: __virtual__ return err msg.

Updated message in test_virtual module when return False.

abednarik committed:
modules.system: __virtual__ return err msg.

Updated message in system  module when return False if system is not Linux or shutdown command is not available.

abednarik committed:
modules.system_profiler: __virtual__ return err msg.

Updated message in system  module when return False if system_profiler  is not available.

immunda committed:
Improve fast-forward error message

xenophonf committed:
Fix a typo in the gpgkey URL

The-Loeki committed:
Enhancement: Add cpu_baseline to virt

This function leverages libvirt to generate a custom CPU map with the optimal settings for a VM on a hypervisor minion

Commit:
Add more virtual return fixes.

Commit:
Add more virtual return fixes.

Commit:
Merge pull request #29793 from abednarik/virtual_ret_system_profiler_module

modules.system_profiler: __virtual__ return err msg.

Commit:
Merge pull request #29791 from abednarik/virtual_ret_test_virtual_module

modules.test_virtual: __virtual__ return err msg.

Commit:
Merge pull request #29788 from abednarik/virtual_ret_win_autoruns_module

modules.win_autoruns: __virtual__ return err msg.

Commit:
Merge pull request #29794 from immunda/patch-2

Improve git fast-forward error message

terminalmage committed:
Remove references to repoquery from comments

Since repoquery is no longer used, the comments should not reference it.

cachedout commented:
Really nice! I like the way you've phrased all the error messages here. They are very clean and concise.

The-Loeki committed:
lintfix

tbaker57 commented:
I'll fix up those lint errors ....

Commit:
Lint fixes

Commit:
Lint fixes

abednarik commented:
Wow! Many thanks for that! Please is there are lint error, let me know. My
idea is to help you out, not to give you more work to do ;) Will check what
those lint errors to avoid having the same mistake.

BTW: In the list, there are lot of tasks that are already en develop branch
but documentation show that there are pending. Do you guys want to remove
that or search each commit and update the list?

Cheers.

2015-12-17 13:45 GMT-03:00 tbaker57 <notifications@github.com>:

> I'll fix up those lint errors ....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saltstack/salt/pull/29798#issuecomment-165505675>.
>

tbaker57 commented:
I think the list on the wiki might not be complete (pinging @jfindlay for confirmation?). I'm using
```
find . -name "*.py" -exec grep -lPzo '(?s)def __virt((?!def).)*return ((?!def).)* False' {} \;
```
which finds (with a couple of false positives) all .py files with virtual return functions without a detailed error msg.

cachedout commented on salt/modules/debian_service.py:
```diff
@@ -36,7 +36,8 @@ def __virtual__():
     if __grains__['os'] in ('Debian', 'Raspbian', 'Devuan') and not salt.utils.systemd.booted(__context__):
         return __virtualname__
-    return False
+    return (False, 'The debian_service module failed to load: only works on Debian, Raspbian or Devuan ' 
```
Please remove the trailing whitespace here.

Commit:
lint - remove trailing whitespace.

Commit:
lint - remove trailing whitespace.

abednarik committed:
modules.pacman: __virtual__ return err msg.

Updated message in pacman module when return False if OS family is not supported.

Commit:
Merge pull request #29782 from akissa/add-pillar-sqlcipher-module

Add Pillar SQLCipher module

Commit:
Merge pull request #29797 from The-Loeki/patch-1

Enhancement: Add cpu_baseline to virt

Commit:
Merge pull request #29799 from terminalmage/remove-repoquery

Remove references to repoquery from comments

Commit:
Merge pull request #29777 from TravisPaul/develop

The cron module does not work with SmartOS

abednarik committed:
modules.freebsdkmod: __virtual__ return err msg.

Updated message in freebsdkmod module when return False if system is not FreeBSD.

Commit:
Merge pull request #29802 from abednarik/virtual_ret_freebsdkmod_module

modules.freebsdkmod: __virtual__ return err msg.

Commit:
Merge pull request #29732 from terminalmage/issue896

Improve error reporting in pkg states

Commit:
Merge pull request #29787 from abednarik/virtual_ret_win_dacl_module

modules.win_dacl: __virtual__ return err msg.

cachedout commented:
Looks like there may be duplicated effort, since this will no longer merge cleanly. Sorry to bother you, @tbaker57 but would you mind rebasing this so I can get it to merge? Thanks.

akissa committed:
add versionadded as requested

Fixes #29781

Commit:
Merge pull request #29807 from cachedout/lint_29686

Lint 29686

Commit:
Merge pull request #29804 from irtnog/fix-rhel-repo-config

Fix a typo in the gpgkey URL

Commit:
Merge pull request #29805 from akissa/add-versionadd-to-pillar-sqlcipher

add versionadded as requested

Commit:
Merge pull request #29801 from abednarik/virtual_ret_pacman_module

modules.pacman: __virtual__ return err msg.

Commit:
Merge pull request #29792 from abednarik/virtual_ret_system_module

modules.system: __virtual__ return err msg.

Commit:
Merge branch 'virt_return_fixes' of github.com:tbaker57/salt into virt_return_fixes

Conflicts:
	salt/modules/freebsdkmod.py

tbaker57 commented:
Whoah! Not sure where all those merge commits came from ? Happy to close this pull request now and start from scratch....

cachedout commented:
Probably easiest to just open a new PR then. That's no problem.
